conda/shell/etc/profile.d/conda.sh

```diff
@@ -84,17 +84,17 @@ conda() {
     fi
     case "$cmd" in
         activate)
-            _conda_activate "$@"
+            _conda_activate "$@" || return $?
             ;;
         deactivate)
-            _conda_deactivate "$@"
+            _conda_deactivate "$@" || return $?
             ;;
         install|update|uninstall|remove)
-            $_CONDA_EXE "$cmd" "$@"
-            _conda_reactivate
+            $_CONDA_EXE "$cmd" "$@" || return $?
```

Comment:
I think only this line should be changed and I'd rather change it to
```bash
            $_CONDA_EXE "$cmd" "$@" && _conda_reactivate
```
For the `eval` lines a `|| return $?` would be rather arbitrary as only the last evaluated command will affect `||`.
For all `case`s , their return codes should be returned anyway as those are the last evaluated commands (nothing else follows `esac`).
(This also implies that for `(de)activate` the return codes are actually the ones from `_conda_hashr`..)